streaming/src/main/scala/com/stratio/crossdata/streaming/helpers/CrossdataStreamingHelper.scala

```diff
@@ -0,0 +1,138 @@
+/**
+ * Copyright (C) 2015 Stratio (http://stratio.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.stratio.crossdata.streaming.helpers
+
+import com.github.nscala_time.time.Imports._
+import com.stratio.common.utils.components.logger.impl.SparkLoggerComponent
+import com.stratio.crossdata.streaming.constants.KafkaConstants._
+import com.stratio.crossdata.streaming.kafka.{KafkaInput, KafkaProducer}
+import com.typesafe.config.{Config, ConfigFactory}
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.DataFrame
+import org.apache.spark.sql.crossdata.XDContext
+import org.apache.spark.sql.crossdata.models._
+import org.apache.spark.streaming.dstream.DStream
+import org.apache.spark.streaming.{Seconds, StreamingContext}
+import org.apache.spark.{SparkConf, SparkContext}
+
+import scala.util.{Failure, Try}
+
+object CrossdataStreamingHelper extends SparkLoggerComponent {
+
+  def createContext(ephemeralTable: EphemeralTableModel,
+                    sparkConf: SparkConf,
+                    zookeeperConf: Map[String, String],
+                    crossdataCatalogConf: Map[String, String]): StreamingContext = {
+    val sparkStreamingWindow = ephemeralTable.options.atomicWindow
+    val sparkContext = SparkContext.getOrCreate(sparkConf)
+    val streamingContext = new StreamingContext(sparkContext, Seconds(sparkStreamingWindow))
+
+    streamingContext.checkpoint(ephemeralTable.options.checkpointDirectory)
+
+    val kafkaOptions = ephemeralTable.options.kafkaOptions
+    val kafkaInput = new KafkaInput(kafkaOptions)
+    val kafkaDStream = toWindowDStream(kafkaInput.createStream(streamingContext), ephemeralTable.options)
+
+    kafkaDStream.foreachRDD { rdd =>
+      if (rdd.take(1).length > 0) {
```

Comment:
Instead of taking 1 row here and collect all the rows later, maybe we could collect all the rows here, thus, 1 output operation would be avoided